completion/available/defaults.completion.bash

```diff
@@ -1,91 +1,122 @@
-# defaults
+# shellcheck shell=bash
+# shellcheck disable=SC2207
+#
 # Bash command line completion for defaults
 #
-# Created by Jonathon Mah on 2006-11-08.
+# Version 1.0 created by Jonathon Mah on 2006-11-08.
 # Copyright 2006 Playhaus. All rights reserved.
 #
-# Version 1.0 (2006-11-08)
 
+shopt -s extglob
 
-_defaults_domains()
+function _defaults_domains()
 {
-    local cur
-    COMPREPLY=()
-    cur=${COMP_WORDS[COMP_CWORD]}
-
-	local domains=$( defaults domains | sed -e 's/, /:/g' | tr : '\n' | sed -e 's/ /\\ /g' | grep "^$cur" )
-	local IFS=$'\n'
-	COMPREPLY=( $domains )
-	if [[ $( echo '-app' | grep "^$cur" ) ]]; then
+	local IFS=$'\n' # Treat only newlines as delimiters in string operations.
+	local LC_CTYPE='C' # Do not consider character set in string operations.
+	local cur="${COMP_WORDS[COMP_CWORD]}"
+	COMPREPLY=()
+
+	#local domains="$( defaults domains )"
+	#local candidates=( $( compgen -W "${domains//, /$'\n'}" | grep -i "^${cur}" ) )
+	local domains="$( defaults domains | sed -e 's/, /^/g' | tr '^' '\n' )"
```

Comment:
It is possible to have a domain named with a comma and a space...so this will simply never be perfect short of implementing a whole entire new parsing/search engine to enumerate defaults domains.